.gitlab-ci.yml

```diff
@@ -428,19 +428,41 @@ check-polkadot-companion-status:
   script:
     - ./.maintain/gitlab/check_polkadot_companion_status.sh
 
-check-polkadot-companion-build:
-  stage:                           build
+.check-companion-build:            &check-companion-build
+  stage:                           test
   <<:                              *docker-env
   <<:                              *test-refs-no-trigger
-  needs:
-    - job:                         test-linux-stable-int
-      artifacts:                   false
   script:
-    - ./.maintain/gitlab/check_polkadot_companion_build.sh
+    - ./.maintain/gitlab/check_companion_build.sh "$COMPANION_ORG" "$COMPANION_REPO" "$COMPANION_BUILD" "$COMPANION_DEPENDENCY"
   after_script:
-    - cd polkadot && git rev-parse --abbrev-ref HEAD
+    - cd "$COMPANION_REPO" && git rev-parse --abbrev-ref HEAD
   allow_failure:                   true
 
+check-polkadot-companion-build:
+  <<: *check-companion-build
+  variables:
+    COMPANION_ORG: paritytech
+    COMPANION_REPO: polkadot
+    COMPANION_DEPENDENCY: paritytech/grandpa-bridge-gadget
+  allow_failure: true
+
+check-cumulus-companion-build:
+  <<: *check-companion-build
+  variables:
+    COMPANION_ORG: paritytech
+    COMPANION_REPO: cumulus
+    COMPANION_BUILD: "cargo check"
```

Comment:
Hmm this isn't really great :P I can understand why you are doing this, but honestly I don't know... 

Overall it would be nice to have pre merge checks that runs the entire test suite of cumulus to make sure we don't break anything..

Reply:
You mean it's better to trigger the entire cumulus CI here? Is it worth doing on every commit? Maybe on some certain files/dirs changes at least?

Reply:
I mean for cumulus it is not that important at the moment. A `cargo test --all --release` would do it as well. However, even that takes some time. I would actually like to have this as a precommit check, but we don't have support for this :(

Reply:
yeah, we're kinda stuck between not being able to use the existing tools for pre-merge pipelines and the inability to use some kind of a pre-commit tool with github.com. This means we'll be looking into how to deal with it nicely, but currently, we're scaling vertically.

Reply:
I suggest we could generalize these checks by actually triggering the CIs of the relevant repository. As we've been doing it [with Simnet](https://github.com/paritytech/polkadot/pull/3151/files) recently. This script passes overriding variables to the downstream CI and returns the status of all checks.
It will require some changes in the relevant repos, (with which @paritytech-ci will certainly help), but will be more future-proof.

Reply:
Now, this triggering script looks [like this](https://github.com/paritytech/polkadot/blob/master/scripts/gitlab/trigger_pipeline.sh), it's a bit more specialized for Simnet.